feat(curriculum): add workshop python functions (#68279)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-kitchen-inventory-tracker/6a3e7284e89ce3367e709331.md

---
id: 6a3e7284e89ce3367e709331
title: Step 13
challengeType: 20
dashedName: step-13
---

# --description--

Now you are going to solve the mystery from earlier! When your `use_eggs` function returned the remaining number of eggs, you didn't save that new value anywhere. It just disappeared into the void! 

> To actually update your inventory, you need to capture that returned value by assigning it to a variable. 

Reviewer: I don't think this is the right position for these 2 paragraphs because, at this stage, you are not solving the problem yet. They would be more suitable for step 14, so please use something else here and move them there.

> Inside your `make_fried_egg` function, replace the `pass` keyword. First, create a variable named `has_enough_eggs` and assign it the result of checking if `available_eggs` is greater than or equal to 1.

Reviewer: ```suggestion
Inside your `make_fried_egg` function, replace the `pass` keyword by creating a variable named `has_enough_eggs` and assigning it the result of checking if `available_eggs` is greater than or equal to 1.
```

> Inside your `make_fried_egg` function, replace the `pass` keyword. First, create a variable named `has_enough_eggs` and assign it the result of comparing if `available_eggs` is greater than or equal to 1.

Reviewer: ```suggestion
Inside your `make_fried_egg` function, replace the `pass` keyword. First, create a variable named `has_enough_eggs` and assign it the result of checking if `available_eggs` is greater than or equal to 1.
```

You should assign `available_eggs` to the result of `use_eggs(eggs, 1)` inside your `make_fried_egg` function.

```js
({
  test: () => assert(
    runPython(`_Node(_code).find_function("make_fried_egg").find_variables("available_eggs")[0].is_equivalent("available_eggs = use_eggs(eggs, 1)")`)
  )
})
```

You should assign `has_enough_eggs` to the result of the comparison `available_eggs >= eggs` inside your `make_fried_egg` function.

```js
({
  test: () => assert(
    runPython(`_Node(_code).find_function("make_fried_egg").find_variables("has_enough_eggs")[0].is_equivalent("has_enough_eggs = available_eggs >= eggs")`)
  )
})
```

# --seed--

## --seed-contents--

```py
eggs = 1
flour = 2
sugar = 3

def take_inventory():
    total_items = eggs + flour + sugar
    print(f'The kitchen has {total_items} total items:')
    print(f'- {eggs} eggs')
    print(f'- {flour} flour')
    print(f'- {sugar} sugar')

take_inventory()

def use_eggs(eggs, num_eggs):
    if num_eggs > eggs:
        print('The kitchen does not have enough eggs.')
        return eggs

    print(f'{num_eggs} egg(s) used out of {eggs} available.')
    return eggs - num_eggs

use_eggs(eggs, 1)
print(eggs)
        

> def make_fried_egg(eggs):        

Reviewer 2: ```suggestion
def make_fried_egg(eggs):
```

Author: neither here

Reviewer 2: trailing whitespaces
